tests/rptest/services/cloud_cluster_utils.py

```diff
@@ -53,6 +53,11 @@ def __init__(self, context, logger, infra_id, infra_secret, provider,
             self.gcp_project_id = self._get_gcp_project_id(infra_id)
             self.logger.info(f"Using GCP project '{self.gcp_project_id}'")
             self.env.update({"GOOGLE_APPLICATION_CREDENTIALS": infra_id})
+        elif self.provider == 'azure':
+            self.subscription_id = context.globals.get('azure_subscription_id',
```

Comment:
`get()` defaults to returning `None` if not found, so `, None` is redundant here, but more important why do we set it to None if not found? Later on we use it w/o checking if `None` so seems unsafe and that we rely on it being not `None`. So we can fail fast here using `[]` for the lookup or else if we only need it later _sometimes_ do additional checking at that point.